zerver/lib/bugdown/fenced_code.py

```diff
@@ -156,13 +163,16 @@ def done(self) -> None:
         raise NotImplementedError()
 
 def generic_handler(processor: Any, output: MutableSequence[str],
-                    fence: str, lang: str,
+                    fence: str, lang: str, header: str,
                     run_content_validators: Optional[bool]=False,
                     default_language: Optional[str]=None) -> BaseHandler:
+    lang = lang.lower()
     if lang in ('quote', 'quoted'):
         return QuoteHandler(processor, output, fence, default_language)
     elif lang == 'math':
         return TexHandler(processor, output, fence)
+    elif lang.startswith('spoiler') or lang.startswith('zpoiler'):
```

Comment:
Does this need to be `startswith`, or can we check for equality here?

Reply:
Whoops, this is a vestige from an earlier approach we had for same-line spoilers; equality should be fine.